main/src/main/java/cgeo/geocaching/CacheDetailActivity.java

```diff
@@ -448,7 +455,9 @@ private void startOrStopGeoDataListener(final boolean initial) {
         } else {
             start = initial;
         }
-        if (start) {
+        if (start && isLiveDetail) {
+            geoDataDisposable.add(locationUpdater.start(GeoDirHandler.UPDATE_GEODATA | GeoDirHandler.UPDATE_DIRECTION));
+        } else {
```

Comment:
This is wrong. The ELSE path would execute for start=false which is not intended

Instead, use just GeoDirHandler.UPDATE_GEODIR